fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -681,7 +682,43 @@ private ParameterDAO makeParameterDAO(Connection connection) throws FHIRPersiste
                         }
                     } else {
                         unsortedResultsList = resourceDao.search(query);
-                        resources = this.convertResourceDTOList(unsortedResultsList, resourceType, elements);
+                        List<com.ibm.fhir.persistence.jdbc.dto.Resource> matchResultList = unsortedResultsList;
+                        List<com.ibm.fhir.persistence.jdbc.dto.Resource> includeResultList = new ArrayList<>();
+
+                        // Check if _include or _revinclude search. If so, remove duplicates from 'include' resources
+                        // (duplicates of both 'match' and 'include' resources) and make sure _elements processing
+                        // is not done for 'include' resources.
+                        if (searchContext.hasIncludeParameters() || searchContext.hasRevIncludeParameters()) {
```

Comment:
Unless it's mentioned elsewhere (or well-known), I didn't initially know that all the matches come first in the returned search results, and are then followed by the includes.  Maybe a quick comment mentioning that, which explains why the code assumes that order.

Reply:
Done.